intellij/src/saros/intellij/editor/EditorManager.java

```diff
@@ -378,9 +383,19 @@ private void sendSelectionInformation(
    * @param project the added project
    */
   private void addProjectResources(IProject project) {
-    Project intellijProject = project.adaptTo(IntelliJProjectImpl.class).getModule().getProject();
+    Module module = project.adaptTo(IntelliJProjectImpl.class).getModule();
+    FileIndex moduleFileIndex = ModuleRootManager.getInstance(module).getFileIndex();
+    Project intellijProject = module.getProject();
 
-    VirtualFile[] openFiles = ProjectAPI.getOpenFiles(intellijProject);
+    Set<VirtualFile> openFiles = new HashSet<>();
+
+    for (VirtualFile openFile : ProjectAPI.getOpenFiles(intellijProject)) {
+      if (moduleFileIndex.isInContent(openFile)) {
```

Comment:
What does the `isInContent` check do? Why do we care for this property for the `openFiles`-set?
I assume this is needed to be able to send the info of the `openFileMapping`-entries, which is built from `openFiles`?

Reply:
`isInContent` checks whether the passed file is part of the content of the module the file index belongs to, i.e. whether the resource belongs to the module. I added this check to filter out any open shared files that belong to a different module. Technically, this is not needed yet, but it is a nice and simple optimization to avoid sending user editor state information for editors that were already shared beforehand as this information should already be available to all participants.